test/controllers/register/trustees/individual/TrusteeNameControllerSpec.scala

```diff
@@ -0,0 +1,155 @@
+      contentAsString(result) mustBe viewAsString()
+    }
+
+    "populate the view correctly on a GET when the question has previously been answered" in {
+      val validData = Json.obj(
+        TrusteesId.toString -> Json.arr(
+          Json.obj("trustee" -> Json.obj(
+            TrusteeNameId.toString -> personName
+          ))
+        )
+      )
+
+      val getRelevantData = new FakeDataRetrievalAction(Some(validData))
+
+      val result = controller(getRelevantData).onPageLoad(NormalMode, firstTrusteeIndex, None)(fakeRequest)
+
+      contentAsString(result) mustBe viewAsString(form.fill(PersonName("John", "Doe")))
+    }
+
+    "redirect to the next page when valid data is submitted" in {
+      val validData = Json.obj(
+        TrusteesId.toString -> Json.arr(
+          Json.obj("trustee" -> Json.obj(
+            TrusteeNameId.toString -> personName
+          ))
+        )
+      )
+
+      when(mockUserAnswersService.upsert(any(), any(), any())(any(), any(), any())).thenReturn(Future.successful(validData))
+
+      val result = controller().onSubmit(NormalMode, firstTrusteeIndex, None)(postRequest)
+
+      status(result) mustBe SEE_OTHER
+    }
+
+    "return a Bad Request and errors when invalid data is submitted" in {
+      val postRequest = fakeRequest.withFormUrlEncodedBody(("firstName", "01"), ("lastName", "?&^%$£"))
+
+      val result = controller().onSubmit(NormalMode, firstTrusteeIndex, None)(postRequest)
+
+      status(result) mustBe BAD_REQUEST
+    }
+
+    "return a Bad Request and errors when no data is submitted" in {
+      val result = controller().onSubmit(NormalMode, firstTrusteeIndex, None)(fakeRequest)
+
+      status(result) mustBe BAD_REQUEST
+    }
+
+    "save the IsTrusteeNewId flag and when the new trustee is being added" in {
+      reset(mockSectionComplete, mockUserAnswersService)
+
+      val validData =
+        UserAnswers().set(TrusteeNameId(firstTrusteeIndex))(PersonName("testFirstName", "testLastName")).flatMap(
+          _.set(IsTrusteeNewId(firstTrusteeIndex))(true)).asOpt.value.json
+
+      val getRelevantData = new FakeDataRetrievalAction(Some(validData))
+
+      when(mockUserAnswersService.upsert(any(), any(), any())(any(), any(), any())).thenReturn(Future.successful(validData))
+
+      val result = controller(getRelevantData).onSubmit(NormalMode, firstTrusteeIndex, None)(postRequest)
+
+      status(result) mustBe SEE_OTHER
+
+      verify(mockUserAnswersService, times(1)).upsert(eqTo(NormalMode), eqTo(None), eqTo(validData))(any(), any(), any())
+    }
+  }
+}
+
+object TrusteeNameControllerSpec extends ControllerSpecBase with MockitoSugar {
+  def onwardRoute: Call = controllers.routes.IndexController.onPageLoad()
+
+  private val formProvider: PersonNameFormProvider = new PersonNameFormProvider()
+  private val form: Form[PersonName] = formProvider()
+
+  private val firstTrusteeIndex: Index = Index(0)
+  private val personName = PersonName("John", "Doe")
```

Comment:
Should we write a generator for `PersonName` for data setup like this? It would allow us to no have hardcoded data.